Upgrade the front-end build tool from roadhog to UmiJS

gurbirkalsi opened the pull request:
This migration resolves issues between differing babel versions for npm submodules and upgrades the routing system. 

The migration consists of the following changes: 
- Upgrade roadhog to umi in `package.json`
- Migrate `.webpackrc.js` and `.babelrc.js` to `config/config.js` for webpack configuration
- Migrate `src/commmon/router.js` to `config/router.config.js`
- Rename `src/routes` to `src/pages`
- Rename` index.js` and `.less` entry points to `global.js` and `.less`
- Add dev config to `mock/datastoreConfig.js` for local dev environment
- Update `services/global.js` to query dev config from `/dev/` endpoint
- Add umi related files to `.gitignore` 
- Modify the component nesting syntax by referencing `{this.props.children}` where `<Switch>` was used

portante commented:
And what about `dev.config.json`?  Are we still using that?  I was not able to get a local dev environment to work.

gurbirkalsi commented:
> Seems like we still have warnings:

@portante Unfortunately, some of these warnings seem to be edge cases of the package manager being used (either npm or yarn). After including the following dependencies in the peerDepedencies{} definition under `package.json` we still see the incorrect peer depenency warning during install. This seems to be a common issue with yarn logging incorrect warnings (however, this is not the case with npm):

portante commented:
@gurbirkalsi, great, can you document that in the README so that people know what to expect as normal when they build?

Should say that "`npm`" is the measure for now, and that folks can use "`yarn`" as well?

gurbirkalsi commented:
> And what about `dev.config.json`? Are we still using that? I was not able to get a local dev environment to work.

`dev.config.json` is not required for local development anymore - I updated the `README.md` to reflect this and updated the doc with new instructions.

With the new routing system, the dashboard will listen to any exported static data defined in the `mock/` directory which can be queried from the global service. Once you've updated the `mock/datastoreConfig.js` file with the proper values, it should initialize the datastore properly within the redux store.

gurbirkalsi commented:
Just updated the `README.md` - let me know if these changes build on your local environment.

portante commented on web-server/v0.4/README.md:
```diff
@@ -63,16 +68,16 @@ This will automatically open the application on [http://localhost:8000](http://l
-Create and include a JSON config file in the root directory. Please reference the following example for required configuration fields.
-
-`dev.config.json`
+Modify the `datastoreConfig.js` file in the mock directory. Please reference the following example for required configuration fields.
```
This is likely going to cause folks a problem, because a person can easily check-in a fully populated `datastoreConfig.js` file when they shouldn't.

Can we provide just a `datastoreConfig.js.example` which we tell people to copy to `datastoreConfig.js` where we add an explicit entry to ignore that file in `.gitignore`?

gurbirkalsi replied:
Yes! This would be a much better approach.

portante commented:
These modules appear to generate NPM warnings still:

```
react-router-redux@5.0.0-alpha.9
circular-json@0.3.3
acorn-jsx@5.0.1
react-fittext@1.0.0
```

Full text of a local build:

```
$ npm install
npm WARN deprecated react-router-redux@5.0.0-alpha.9: This project is no longer maintained.
npm WARN deprecated circular-json@0.3.3: CircularJSON is in maintenance only, flatted is its successor.

> fsevents@1.2.7 install /Users/portante/Documents/DSA/pbench/tree-dsa/web-server/v0.4/node_modules/fsevents
> node install

node-pre-gyp WARN Using request for node-pre-gyp https download 
[fsevents] Success: "/Users/portante/Documents/DSA/pbench/tree-dsa/web-server/v0.4/node_modules/fsevents/lib/binding/Release/node-v67-darwin-x64/fse.node" is installed via remote

> husky@0.14.3 install /Users/portante/Documents/DSA/pbench/tree-dsa/web-server/v0.4/node_modules/husky
> node ./bin/install.js

husky
setting up Git hooks
done


> puppeteer@1.12.2 install /Users/portante/Documents/DSA/pbench/tree-dsa/web-server/v0.4/node_modules/puppeteer
> node install.js
.
.
.
Chromium downloaded to /Users/portante/Documents/DSA/pbench/tree-dsa/web-server/v0.4/node_modules/puppeteer/.local-chromium/mac-624492

> preact@8.3.1 postinstall /Users/portante/Documents/DSA/pbench/tree-dsa/web-server/v0.4/node_modules/preact
> node -e "console.log('\u001b[35m\u001b[1mLove Preact? You can now donate to our open collective:\u001b[22m\u001b[39m\n > \u001b[34mhttps://opencollective.com/preact/donate\u001b[0m')"

Love Preact? You can now donate to our open collective:
 > https://opencollective.com/preact/donate
npm notice created a lockfile as package-lock.json. You should commit this file.
npm WARN react-fittext@1.0.0 requires a peer of react@^15.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN react-fittext@1.0.0 requires a peer of react-dom@^15.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN acorn-jsx@5.0.1 requires a peer of acorn@^6.0.0 but none is installed. You must install peer dependencies yourself.

added 3054 packages from 1174 contributors and audited 128061 packages in 66.102s
found 1 low severity vulnerability
  run `npm audit fix` to fix them, or `npm audit` for details
```

portante commented on web-server/v0.4/README.md:
```diff
@@ -63,16 +68,16 @@ This will automatically open the application on [http://localhost:8000](http://l
-Create and include a JSON config file in the root directory. Please reference the following example for required configuration fields.
-
-`dev.config.json`
+Rename the `datastoreConfig.js.example` file in the `mock/` directory to `datastoreConfig.js` and modify the configuration fields within the route definition. Please reference the following example for required configuration fields.
```
Perhaps use "Copy" instead of "Rename"?
